src/GCHP_GridComp/GCHPctmEnv_GridComp/GCHPctmEnv_GridCompMod.F90

```diff
@@ -672,343 +464,132 @@ subroutine Run ( GC, IMPORT, EXPORT, CLOCK, RC )
          _VERIFY(STATUS)
          ALLOCATE(VCr8(is:ie,js:je,lm), STAT=STATUS); 
          _VERIFY(STATUS)
-         ALLOCATE(PLEr8(is:ie,js:je,lm+1), STAT=STATUS); 
-         _VERIFY(STATUS)
-
+         
          ! Prepare inputs to fv_computeMassFluxes
-
-         ! Copy UA,VA to UC,VC while handling top-down vs. bottom-up meteorological data
          if (meteorology_vertical_index_is_top_down) then
-            UC(:,:,:) = UA(:,:,LM:1:-1)
-            VC(:,:,:) = VA(:,:,LM:1:-1)
+            UC(:,:,:) = UA_IMPORT(:,:,:)
+            VC(:,:,:) = VA_IMPORT(:,:,:)
          else
-            UC(:,:,:) = UA(:,:,:)
-            VC(:,:,:) = VA(:,:,:)
+            UC(:,:,:) = UA_IMPORT(:,:,LM:1:-1)
+            VC(:,:,:) = VA_IMPORT(:,:,LM:1:-1)
          end if
+         
          ! Restagger winds (A-grid to C-grid)
-         call A2D2C(U=UC, V=VC, npz=lm, getC=.true.)
-
-         ! fv_computeMassFluxes expects top-down indexing
-         UC(:,:,:) =  UC(:,:,LM:1:-1)
-         VC(:,:,:) =  VC(:,:,LM:1:-1)
+         call A2D2C(U=UC, V=VC, npz=lm, getC=.true.) ! real4 only
+         
          ! Convert real4->real8 for fv_computeMassFluxes 
-         UCr8  = 1.00d0*(UC)
-         VCr8  = 1.00d0*(VC)
-
-         PLEr8 = 1.00d0*(PLE0r8)
-
+         UCr8  = dble(UC)
+         VCr8  = dble(VC)
+         
 #ifdef ADJOINT
          if (.not. firstRun) THEN
 #endif
-         ! Calculate mass fluxes and courant numbers
-         call fv_computeMassFluxes(UCr8, VCr8, PLEr8, &
-                                   MFX, MFY, CX, CY, dt)
+            ! Calculate mass fluxes and courant numbers
+            call fv_computeMassFluxes(UCr8, VCr8, PLE, &
```

Comment:
What was the rational for keeping r8 in the varnames for UC/VC but removing PLE?

Reply:
Just copy pasta!